scripts/nbconverted/process_xCell.py

```diff
@@ -0,0 +1,145 @@
+
+# coding: utf-8
+
+# # Process xCell cell-type genesets
+# 
+# The genesets were extracted from additional file 3 in [Aran et al. 2017](https://doi.org/10.1186/s13059-017-1349-1 "xCell: digitally portraying the tissue cellular heterogeneity landscape"). The data consist of 489 gene signatures. Many of these signatures represent redundant cell-types with different evidence.
+
+# In[1]:
+
+
+import os
+from datetime import datetime
+import csv
+import pandas as pd
+
+
+# In[2]:
+
+
+get_ipython().run_line_magic('matplotlib', 'inline')
+
+
+# In[3]:
+
+
+# Load Additional File 3
+xcell_file = os.path.join('data', '13059_2017_1349_MOESM3_ESM.xlsx')
+xcell_df = pd.read_excel(xcell_file, index_col=0)
+print(xcell_df.shape)
+xcell_df.head()
+
+
+# In[4]:
+
+
+# Distribution of number of cell types
+cell_types = pd.Series([x.split('_')[0] for x in xcell_df.index])
+cell_types.value_counts().head(10)
+
+
+# In[5]:
+
+
+# Distribution of number of cell types
+cell_types.value_counts().hist(bins=10);
+
+
+# In[6]:
+
+
+# Load curated gene names from versioned resource 
+commit = '721204091a96e55de6dcad165d6d8265e67e2a48'
+url = 'https://raw.githubusercontent.com/cognoma/genes/{}/data/genes.tsv'.format(commit)
+gene_df = pd.read_table(url)
+
+# Only consider protein-coding genes
+gene_df = (
+    gene_df.query("gene_type == 'protein-coding'")
+)
+
+symbol_to_entrez = dict(zip(gene_df.symbol,
+                            gene_df.entrez_gene_id))
+
+
+# In[7]:
+
+
+# Add alternative symbols to entrez mapping dictionary
+gene_df = gene_df.dropna(axis='rows', subset=['synonyms'])
+gene_df.synonyms = gene_df.synonyms.str.split('|')
+
+all_syn = (
+    gene_df.apply(lambda x: pd.Series(x.synonyms), axis=1)
+    .stack()
+    .reset_index(level=1, drop=True)
+)
+
+# Name the synonym series and join with rest of genes
+all_syn.name = 'all_synonyms'
+gene_with_syn_df = gene_df.join(all_syn)
+
+# Remove rows that have redundant symbols in all_synonyms
+gene_with_syn_df = (
+    gene_with_syn_df
+    
+    # Drop synonyms that are duplicated - can't be sure of mapping
+    .drop_duplicates(['all_synonyms'], keep=False)
+
+    # Drop rows in which the symbol appears in the list of synonyms
+    .query('symbol not in all_synonyms')
```

Comment:
Want to make sure that you're not filtering out approved symbols that appear somewhere as synonyms. You should drop synonyms that are also approved symbols... but approved symbols should always map.

Reply:
> Want to make sure that you're not filtering out approved symbols that appear somewhere as synonyms. You should drop synonyms that are also approved symbols... but approved symbols should always map.

Correct - In [line 61](https://github.com/greenelab/interpret-compression/pull/6/files/e9f81a364c91ab3aca3037d55d4b3b472a27b012#diff-4b2cf60d46ef2ff079e5afeec51e6506R61) I create the initial mapping of symbol to entrez_id. The next mapping ([line 98](https://github.com/greenelab/interpret-compression/pull/6/files/e9f81a364c91ab3aca3037d55d4b3b472a27b012#diff-4b2cf60d46ef2ff079e5afeec51e6506R98)) maps synonym to entrez_id. The line in question (line 90) drops rows where a synonym matches any symbol. 
 
The `symbol_to_entrez` dictionary is then updated to include these synonym to entrez mappings ([line 101](https://github.com/greenelab/interpret-compression/pull/6/files/e9f81a364c91ab3aca3037d55d4b3b472a27b012#diff-4b2cf60d46ef2ff079e5afeec51e6506R101))